CMakeLists.txt

```diff
@@ -56,78 +56,131 @@ if( ENABLE_TESTING )
     find_package( GTest REQUIRED )
 
     set ( TEST_DRIVE_FIND_METHOD fetch )
-    # option TEST_DRIVE_BUILD_TESTING is hard-coded to ON
+    # Option TEST_DRIVE_BUILD_TESTING is hard-coded to ON, could be spared
     find_package( test-drive REQUIRED )
 endif()
 
+# Optionally use PackageCompiler.jl to build standalone libtrixi.so
+option( USE_PACKAGE_COMPILER "Build standalone libtrixi.so using PackageCompiler.jl" )
 
+# Fortran mod file location
+set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR})
 
-# Library target
-add_library ( ${PROJECT_NAME} SHARED
-    src/api.c
-    src/api.f90
-    src/auxiliary.h
-    src/auxiliary.c
-    src/trixi.h
-)
+if( USE_PACKAGE_COMPILER )
+    if ( NOT DEFINED JULIA_PROJECT_PATH )
+        message( FATAL_ERROR "JULIA_PROJECT_PATH needs to be set for PackageCompiler.jl.")
+    endif()
+    if ( ENABLE_TESTING )
+        message( NOTICE "Testing is not supported when PackageCompiler is used.")
+    endif()
 
-# Include directories, private
-target_include_directories ( ${PROJECT_NAME} PRIVATE src )
+    # Copy source with initialization routines; an object file will be created
+    file( MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/build-pc )
+    file( COPY_FILE ${CMAKE_SOURCE_DIR}/LibTrixi.jl/lib/init.c
+                    ${CMAKE_BINARY_DIR}/build-pc/init.c
+                    ONLY_IF_DIFFERENT )
+
+    # Define PackageCompiler.jl output file
+    set( PC_LIBTRIXI_SO ${CMAKE_BINARY_DIR}/prefix-pc/lib/libtrixi.so )
+
+    # Add a library target (libtrixi), only for Fortran module
+    add_library( ${PROJECT_NAME} OBJECT
+        src/api.f90
+    )
+
+    # Custom command to run PackageCompiler.jl to produce libtrixi.so
+    add_custom_command( OUTPUT ${PC_LIBTRIXI_SO}
+                        COMMENT "Building ${PROJECT_NAME} with PackageCompiler.jl..."
+                        COMMAND ${JULIA_EXECUTABLE}
+                                --project=${CMAKE_SOURCE_DIR}/LibTrixi.jl/lib
+                                ${CMAKE_SOURCE_DIR}/LibTrixi.jl/lib/build.jl
+                                ${JULIA_PROJECT_PATH}
+                                ${CMAKE_BINARY_DIR}/prefix-pc
+                        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/build-pc )
+
+    # Custom target for PackageCompiler.jl's libtrixi.so
+    add_custom_target( PC_LIBTRIXI DEPENDS ${PC_LIBTRIXI_SO} )
+
+    # Dependency of main library target on PackageCompiler.jl target
+    add_dependencies( ${PROJECT_NAME} PC_LIBTRIXI )
+
+    # Add linking to PackageCompiler.jl's libtrixi.so
+    target_link_libraries( ${PROJECT_NAME} INTERFACE ${PC_LIBTRIXI_SO} )
+
+    # Install configuration
+    install( DIRECTORY "${CMAKE_BINARY_DIR}/prefix-pc/lib/" TYPE LIB )
+    install( DIRECTORY "${CMAKE_BINARY_DIR}/prefix-pc/share/julia"
+             DESTINATION share
+             PATTERN "cert.pem" EXCLUDE )
```

Comment:
It has some certificates required for communicating safely with some of the Julia servers, I believe (e.g., package servers). We should not include it, otherwise we might get hard-to-understand errors...

Reply:
Just to clarify:

> We should not include it, otherwise we might get hard-to-understand errors...

Should we, or should we not?

Reply:
We should not include the exclusion :-) I've already updated the code 😬